api/src/test/java/org/openmrs/api/EncounterServiceTest.java

```diff
@@ -98,7 +97,235 @@
 	public void runBeforeEachTest() throws Exception {
 		executeDataSet(ENC_INITIAL_DATA_XML);
 	}
-	
+
+	@Test
+	@Verifies(value = "should save encounter correctly when a leaf child obs updated without recreating an entire obs hierarchy", method = "saveEncounter(Encounter)")
+	public void saveEncounter_shouldNotCreateACopyOfParentObsButEditChildObs(){
+		ConceptService cs = Context.getConceptService();
+		EncounterService es = Context.getEncounterService();
+		ObsService os = Context.getObsService();
+
+		//create an obs
+		Obs o = createObs(cs.getConcept(3),"original obs value text");
+		//create a second obs
+		Obs o2 = createObs(cs.getConcept(3),"second obs value text");
+		//create a parent obs
+		Obs oParent =createObs(cs.getConcept(23), null); //in the concept set table as a set
+
+		//add o and o2 to the parent obs
+		oParent.addGroupMember(o2);
+		oParent.addGroupMember(o);
+
+		Encounter enc = buildEncounter();
+		enc.addObs(oParent);
+
+		enc = es.saveEncounter(enc);
+
+		//Now updating o
+		o.setValueText("Obs value updated");
+		enc = es.saveEncounter(enc);
+
+		Set<Obs> obsAtTopLevelUpdated = enc.getObsAtTopLevel(true);
+		oParent = os.getObs(oParent.getObsId());
+		o = os.getObs(o.getObsId());
+		o2 = os.getObs(o2.getObsId());
+		Obs o3 = getNewVersionOfEditedObs(oParent,o);
+
+		assertEquals(1,obsAtTopLevelUpdated.size());
+		assertEquals(3, oParent.getGroupMembers(true).size());
+		assertTrue(o.getVoided());
+		assertFalse(oParent.getVoided());
+		assertFalse(o2.getVoided());
+
+		assertNotNull(o3);
+		assertFalse(o3.getVoided());
+		assertEquals("Obs value updated", o3.getValueText());
+	}
+
+	@Test
+	@Verifies(value = "should save encounter correctly when a level 1 child obs updated without recreating an entire obs hierarchy", method = "saveEncounter(Encounter)")
```

Comment:
So this test seems to contain the tree of obs I expected in the other test, when you say level one i would expect editing a child and not a grand parent. I think use child and nest child to be clearer